src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,837 @@
+  }
+
+  @Listen("keydown")
+  keyDownHandler(event: KeyboardEvent): void {
+    switch (this.activeEl) {
+      case this.hourEl:
+        if (event.key === "ArrowRight") {
+          this.setFocus("minute");
+        }
+        break;
+      case this.minuteEl:
+        switch (event.key) {
+          case "ArrowLeft":
+            this.setFocus("hour");
+            break;
+          case "ArrowRight":
+            if (this.step !== 60) {
+              this.setFocus("second");
+            } else if (this.hourDisplayFormat === "12") {
+              this.setFocus("meridiem");
+            }
+            break;
+        }
+        break;
+      case this.secondEl:
+        switch (event.key) {
+          case "ArrowLeft":
+            this.setFocus("minute");
+            break;
+          case "ArrowRight":
+            if (this.hourDisplayFormat === "12") {
+              this.setFocus("meridiem");
+            }
+            break;
+        }
+        break;
+      case this.meridiemEl:
+        switch (event.key) {
+          case "ArrowLeft":
+            if (this.step !== 60) {
+              this.setFocus("second");
+            } else {
+              this.setFocus("minute");
+            }
+            break;
+        }
+        break;
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  @Method()
+  async setFocus(target: TimeFocusId): Promise<void> {
+    this[`${target || "hour"}El`]?.focus();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private decrementHour = (): void => {
+    const newHour = !this.hour ? 0 : this.hour === "00" ? 23 : parseInt(this.hour) - 1;
+    this.setTime("hour", newHour);
+  };
+
+  private decrementMeridiem = (): void => {
+    const newMeridiem = this.meridiem === "PM" ? "AM" : "PM";
+    this.setTime("meridiem", newMeridiem);
+  };
+
+  private decrementMinuteOrSecond = (key: MinuteOrSecond): void => {
+    let newValue;
+    if (isValidNumber(this[key])) {
+      const valueAsNumber = parseInt(this[key]);
+      if (valueAsNumber === 0) {
+        newValue = 59;
+      } else {
+        newValue = valueAsNumber - 1;
+      }
+    } else {
+      newValue = 59;
+    }
+    this.setTime(key, newValue);
+  };
+
+  private decrementMinute = (): void => {
+    this.decrementMinuteOrSecond("minute");
+  };
+
+  private decrementSecond = (): void => {
+    this.decrementMinuteOrSecond("second");
+  };
+
+  private buttonActivated(event: KeyboardEvent): boolean {
+    const isKeyboardEvent = event instanceof KeyboardEvent;
+    const enterPressed = isKeyboardEvent && event.key === "Enter";
+    const spacebarPressed = isKeyboardEvent && (event.key === " " || event.key === "Spacebar");
```

Comment:
For consistent handling of key values, use the `key#getKey` helper: https://github.com/Esri/calcite-components/blob/f8077418e5b13384d62942918ecc70c3de76f9ab/src/utils/key.ts#L4-L3